src/pb_tensor.cc

```diff
@@ -305,12 +348,42 @@ PbTensor::Memory()
 
 #ifdef TRITON_PB_STUB
 std::shared_ptr<PbTensor>
-PbTensor::FromDLPack(const std::string& name, const py::capsule& dlpack_tensor)
+PbTensor::FromDLPack(const std::string& name, const py::object& tensor)
 {
   if (name == "") {
     throw PythonBackendException("Tensor name cannot be an empty string.");
   }
+  if (py::isinstance<py::capsule>(tensor)) {
+    return FromDLPackCapsule(name, tensor);
+  } else if (py::hasattr(tensor, "__dlpack__")) {
+    // Python backend does not support async executions on PbTensor objects,
+    // thus we can count on the default `stream=None` argument.
+    // For CPU, `stream=None` is the only accepted argument
+    // according to array API. For GPU, when `stream=None`  producer must
+    // assume the legacy default stream.
+    // Reference:
+    // https://data-apis.org/array-api/latest/API_specification/generated/array_api.array.__dlpack__.html
+    return FromDLPackCapsule(
+        name, tensor.attr("__dlpack__")("stream"_a = py::none()));
+  } else {
+    throw PythonBackendException(
+        "Provided tensor is not supported.\
+      Tensor must be a DLPack capsule or have a `__dlpack__` attribute");
+  }
+}
 
+std::shared_ptr<PbTensor>
+PbTensor::FromDLPackCapsule(
+    const std::string& name, const py::capsule& dlpack_tensor)
+{
+#ifdef TRITON_ENABLE_GPU
+  cudaError_t err = cudaStreamSynchronize(0);
```

Comment:
We only need the synchronization when the tensor type is `GPU`. We should add a condition around this so that it is executed only on GPU tensors. Also, what happens in the multi-GPU case? Do we need to change the context to the correct GPU before synchronization?